Fix ownership of deployment before before_deploy hook

smortex opened the pull request:
If a `before_deploy` hook has to put files in the deployment directory, ensure it can do so.

neomilium commented:
Could you rebase it?
I may noticed something clever: doing `chown_R` right after extraction makes more sense than this patch combined with `chown_R` after all steps, what do you think about it?
To trig that, I think to `after_deploy` script that need to `chown` some files to give some extra access to tiers.

smortex commented:
Yeah, extracting as root is probably not the best when we have a restricted deploy user. I added in-line comments for the reasons of the steps and added some info on how to improve this.
